### PR TITLE
Build UWP with native UWP rustc target

### PR DESCRIPTION
**Pull request by jdm**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/24198)

### CODE REVIEW

**Comment by jdm** [comment]
```
warning: Patch `rustc-std-workspace-alloc v1.0.0 (C:\Users\task_1568230939\.rustup\toolchains\nightly-2019-08-26-x86_64-pc-windows-msvc\lib\rustlib\src\rust\src\tools\rustc-std-workspace-alloc)` was not used in the crate graph.
Patch `rustc-std-workspace-core v1.0.0 (C:\Users\task_1568230939\.rustup\toolchains\nightly-2019-08-26-x86_64-pc-windows-msvc\lib\rustlib\src\rust\src\tools\rustc-std-workspace-core)` was not used in the crate graph.
Check that the patched package version and available features are compatible
with the dependency requirements. If the patch has a different version from
what is locked in the Cargo.lock file, run `cargo update` to use the new
version. This may also occur with an optional dependency that is not enabled.
   Compiling core v0.0.0 (C:\Users\task_1568230939\.rustup\toolchains\nightly-2019-08-26-x86_64-pc-windows-msvc\lib\rustlib\src\rust\src\libcore)
    Finished release [optimized] target(s) in 14.49s
warning: dependency (compiler_builtins) specified without providing a local path, Git repository, or version to use. This will be considered an error in future versions
    Updating crates.io index
warning: Patch `rustc-std-workspace-alloc v1.0.0 (C:\Users\task_1568230939\.rustup\toolchains\nightly-2019-08-26-x86_64-pc-windows-msvc\lib\rustlib\src\rust\src\tools\rustc-std-workspace-alloc)` was not used in the crate graph.
Patch `rustc-std-workspace-core v1.0.0 (C:\Users\task_1568230939\.rustup\toolchains\nightly-2019-08-26-x86_64-pc-windows-msvc\lib\rustlib\src\rust\src\tools\rustc-std-workspace-core)` was not used in the crate graph.
Check that the patched package version and available features are compatible
with the dependency requirements. If the patch has a different version from
what is locked in the Cargo.lock file, run `cargo update` to use the new
version. This may also occur with an optional dependency that is not enabled.
 Downloading crates ...
  Downloaded compiler_builtins v0.1.19
   Compiling compiler_builtins v0.1.19
    Finished release [optimized] target(s) in 29.28s
```

```
error[E0463]: can't find crate for `std`
  |
  = note: the `aarch64-uwp-windows-msvc` target may not be installed

error: aborting due to previous error
```

**Comment by jdm** [comment]
Note to self: need to get https://github.com/jdm/mozjs/commit/e0adb8cad8224ec9baa5f65558557fb83b7017a0 merged.

**Comment by jdm** [comment]
This got much farther than before, but it appears that [autocfg](https://crates.io/crates/autocfg) is broken in the presence of xargo.

**Comment by jdm** [comment]
Progress! We reached the linking stage in the arm64 build, and then hit:
```
  = note: lld-link: error: could not open opengl32.lib: no such file or directory
```
Sadly, https://github.com/retep998/winapi-rs/blob/b67cb33fa86dba4fb025a339341521660332751d/build.rs#L454 is checking a specific arm64 windows target that we don't match any more.

**Comment by jdm** [comment]
x64 build passed \o/

**Comment by jdm** [comment]
Depends on https://github.com/servo/mozjs/pull/210.

File: Cargo.toml [hunk 1]
```diff
@@ -30,4 +30,10 @@ mio = { git = "https://github.com/servo/mio.git", branch = "servo" }
 iovec = { git = "https://github.com/servo/iovec.git", branch = "servo" }
 rand_os = { git = "https://github.com/servo/rand", branch = "servo-rand_os-0.1.3-uwp" }
 rand_core = { git = "https://github.com/servo/rand", branch = "servo-rand_os-0.1.3-uwp" }
+# Fork with a xargo fix that has not merged upstream yet.
+autocfg = { git = "https://github.com/servo/autocfg", branch = "rustflags" }
+# Fork with a fix for the new arm64 UWP taraget.
+winapi = { git = "https://github.com/jdm/winapi-rs", branch = "patch-1" }
```
Comment (SimonSapin) [inline Cargo.toml]: Could we move this in the servo org?

Comment (SimonSapin) [inline Cargo.toml]: Please link PRs to upstream in those TOML comments

File: Cargo.toml [hunk 2]
```diff
@@ -30,4 +30,10 @@ mio = { git = "https://github.com/servo/mio.git", branch = "servo" }
...
 
+[patch."https://github.com/servo/mozjs"]
+mozjs_sys = { git = "https://github.com/jdm/mozjs", branch = "uwp-rust-changes2" }
```
Comment (SimonSapin) [inline Cargo.toml]: Can this be removed, with https://github.com/servo/mozjs/pull/210 merged?

File: Xargo.toml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+[target.x86_64-uwp-windows-msvc.dependencies]
+std = { features = ["panic-unwind"] }
+
+[target.aarch64-uwp-windows-msvc.dependencies]
+std = { features = ["panic-unwind"] }
```
Comment (SimonSapin) [inline Xargo.toml]: This appears to be undoing https://github.com/rust-lang/rust/blob/000d90b11f7be70ffb7812680f7abc6deb52ec88/src/librustc_target/spec/aarch64_uwp_windows_msvc.rs#L8-L9 , which has a comment that says it shouldn’t be there. Is there a rustc bug we could link here?

File: Xargo.toml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+[target.x86_64-uwp-windows-msvc.dependencies]
+std = { features = ["panic-unwind"] }
```
Comment (SimonSapin) [inline Xargo.toml]: Is this necessary? Unlike the ARM target (see comment below), this appears to default to keep the `PanicStrategy::Unwind` default: https://github.com/rust-lang/rust/blob/000d90b11f7be70ffb7812680f7abc6deb52ec88/src/librustc_target/spec/x86_64_uwp_windows_msvc.rs

File: etc/taskcluster/decision_task.py [hunk 5]
```diff
@@ -393,8 +393,8 @@ def windows_arm64():
         windows_build_task("UWP dev build", arch="arm64", package=False)
         .with_treeherder("Windows arm64")
         .with_script(
-            "python mach build --dev --uwp --win-arm64",
```
Comment (SimonSapin) [inline etc/taskcluster/decision_task.py]: Are the `--uwp` and `--win-arm64` flags still useful, or should they be removed from mach?

File: Xargo.toml (hunk 4 above)
Comment (jdm) [reply Xargo.toml]: Xargo doesn't build libstd successfully unless one of panic-unwind or panic-abort is specified.

File: etc/taskcluster/decision_task.py (hunk 5 above)
Comment (jdm) [reply etc/taskcluster/decision_task.py]: I think it's time to remove them.

**Comment by jdm** [comment]
Review comments addressed.

**Comment by CYBAI** [comment]
```
NuGet Config files used:
    C:\Users\task_1570875509\AppData\Roaming\NuGet\NuGet.Config
    C:\Program Files (x86)\NuGet\Config\Microsoft.VisualStudio.Offline.config

Feeds used:
    https://api.nuget.org/v3/index.json
    C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\

Installed:
    3 package(s) to packages.config projects
error: failed to select a version for `std`.
    ... required by package `sysroot v0.0.0 (C:\Users\TASK_1~2\AppData\Local\Temp\xargo.PnpZ6cqCHEy0)`
versions that meet the requirements `= 0.0.0` are: 0.0.0

the package `sysroot` depends on `std`, with features: `panic-abort` but `std` does not have these features.


failed to select a version for `std` which could resolve this conflict
error: `"cargo" "build" "--release" "--manifest-path" "C:\\Users\\TASK_1~2\\AppData\\Local\\Temp\\xargo.PnpZ6cqCHEy0\\Cargo.toml" "--target" "aarch64-uwp-windows-msvc" "-p" "std"` failed with exit code: Some(101)
stack backtrace:
   0: <no info> (0x7ff635e7d17d)
   1: <no info> (0x7ff635e7c449)
   2: <no info> (0x7ff635e7bf82)
   3: <no info> (0x7ff635e7c023)
   4: <no info> (0x7ff635e4aff8)
   5: <no info> (0x7ff635e3b262)
   6: <no info> (0x7ff635e3ec3b)
   7: <no info> (0x7ff635e2c19e)
   8: <no info> (0x7ff635e31d26)
   9: <no info> (0x7ff635e96d47)
  10: <no info> (0x7ff635ea2572)
  11: <no info> (0x7ff635e975c2)
  12: <no info> (0x7ff635e2d407)
  13: <no info> (0x7ff635eb5864)
  14: BaseThreadInitThunk (0x7ff9561f84d4)
  15: RtlUserThreadStart (0x7ff95657e851)
```

**Pull request merged**